### DISCUSSION TITLE
[V3] Property Reset Doesn't Update HTML

### DISCUSSION THREAD

**Discussion by devajmeireles** [livewire/livewire#5797] [7. Bugs (archived)]
Hello! I'm trying Livewire V3 and I think there is some bug when reset properties.  

# Environment

Laravel: 10.15.0
MacOS: 13.4.1
Livewire: V3

# Code

```php
#[Rule('required|max:100|unique:todos,title')]
public string $title = '';

public function render(): View
{
    return view('livewire.todo.create');
}

public function add(): void
{
    $this->validate();

    Todo::create(['title' => $this->title]);

    $this->dispatch('todo::index::refresh');

    $this->reset('title');
    
    // $this->title = ''; // ◀️  I use this in V2
}
```

# Blade

```php
<div class="m-6">
    <form wire:submit="add" method="post" class="mt-6 space-y-6">
        <div>
            <x-input-label for="name" value="Title"/>

            <x-text-input wire:model="title" id="title" title="title" type="text" class="mt-1 block w-full"/>

            <x-input-error class="mt-2" :messages="$errors->get('title')"/>
        </div>

        <div class="flex items-center justify-end gap-4">
            <x-primary-button>{{ __('Save') }}</x-primary-button>
        </div>
    </form>
</div>
```

# Problem

The field value is not reset visually. Demo: https://share.cleanshot.com/0k0QRlF89yq6pCGjxF6L

**Comment by devajmeireles** [comment]
Comparison between V2 and V3: https://share.cleanshot.com/fZ23LQByRqL3cPLGL8WM

Code 👇

![CleanShot 2023-07-20 at 9 18 52](https://github.com/livewire/livewire/assets/60591772/bf89329b-2729-46a5-b248-be3998ad9d4d)

**Comment by joshhanley** [comment]
@devajmeireles thanks for reporting! Any chance you could submit a PR with a failing test? Thanks!

**Comment by devajmeireles** [comment]
> @devajmeireles thanks for reporting! Any chance you could submit a PR with a failing test? Thanks!

Hey, @joshhanley ! I think I'm not able to write a test due the test schema of the Livewire to be different for me. Could you do it, please?

Have you already tested to check if it happens for you too? I was able to reproduce the same issue with a different computer, Macbook M1 Pro.

**Reply by joshhanley** [reply]
@devajmeireles Why are you not able to write a test? We don't have time at the moment to test out every issue but a failing test makes it easier to review the problem.

If you get the chance, try and see if you can make a failing test for Livewire. It should be pretty straight forward and would be a great help for us. 

I'm happy to help guide you if you run into problems writing the test. Just give it a go and let me know if you have any issues. Also have a look at our contribution guide (for v2 but should be similar for v3). https://laravel-livewire.com/docs/2.x/contribution-guide

Thanks

**Comment by devajmeireles** [comment]
> @devajmeireles Why are you not able to write a test? We don't have time at the moment to test out every issue but a failing test makes it easier to review the problem.
> 
> If you get the chance, try and see if you can make a failing test for Livewire. It should be pretty straight forward and would be a great help for us.
> 
> I'm happy to help guide you if you run into problems writing the test. Just give it a go and let me know if you have any issues. Also have a look at our contribution guide (for v2 but should be similar for v3). https://laravel-livewire.com/docs/2.x/contribution-guide
> 
> Thanks

I was able to write a simple test locally and the test passed successfully: 
![CleanShot 2023-07-22 at 2 28 10](https://github.com/livewire/livewire/assets/60591772/964d5cbf-28d4-4df7-85a2-50979614c749)

To be honest, I don't know what could be the reason for the problem, as the problem occurs on two of my computers with completely different environment settings, but both with PHP 8.1 and Laravel 10.15.

I recorded a video without sound. Could you check if I'm doing something wrong please? To me this is madness. Why would I be wrong if what I do is the same as what I did in the test that passed?

This is the video:
https://www.loom.com/share/28e5799005f847a198e49814d058c456

**Comment by devajmeireles** [comment]
@joshhanley ,

Thank you for all your assistance. I found the reason to this problem, so I decide to create a PR for that: https://github.com/livewire/livewire/pull/5881

**Reply by joshhanley** [reply]
@devajmeireles no worries! Ok great, thanks for that. I will try and get Caleb to have a look at it in the next couple of days 🙂

**Reply by krekas** [reply]
But this PR updated only the docs. Having the same problem where on the frontend still seeing text in the input while property is reseted. The same happens if I use `updated` hook, the property gets updated but in the input it still stays the same.
```php
public function updatedFormTitle($value): void
{
    $this->form->title = Str::headline($value);
}
```

**Reply by joshhanley** [reply]
@krekas are you able to submit a failing test PR for that? Thanks!

**Reply by krekas** [reply]
@joshhanley tried to add them in this PR #6102
